### PR TITLE
Add variable backend validations and checks

### PR DESCRIPTION
**Pull request by shubhamraj-git**
related: #43709 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by shubhamraj-git** [comment]
@jscheffl @pierrejeambrun @bbovenzi 
This PR will handle the validation and checks and will unblock the
https://github.com/apache/airflow/pull/44942

File: airflow/api_fastapi/core_api/routes/public/variables.py [hunk 1]
```diff
@@ -166,6 +167,21 @@ def post_variable(
...
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=f"Variable key exceeds maximum length of {ID_LEN} characters.",
+        )
+
+    # Check if the key already exists
+    existing_variable = session.scalar(select(Variable).where(Variable.key == post_body.key).limit(1))
+    if existing_variable:
+        raise HTTPException(
+            status_code=status.HTTP_409_CONFLICT,
```
Comment (jscheffl) [inline airflow/api_fastapi/core_api/routes/public/variables.py:174]: I think the error 409 need to be added to the decorator to be also added to the OpenAPI spec and therefore to be into the client generated code

File: airflow/api_fastapi/core_api/routes/public/variables.py [hunk 2]
```diff
@@ -166,6 +167,21 @@ def post_variable(
     session: SessionDep,
 ) -> VariableResponse:
     """Create a variable."""
+    # Check for key length limit
+    if len(post_body.key) > ID_LEN:
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=f"Variable key exceeds maximum length of {ID_LEN} characters.",
+        )
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/variables.py]: This shouldn't be done manually. Pydantic have a plethora of native validators for common use cases.
```
Field(max_length=256)
```

File: airflow/api_fastapi/core_api/routes/public/variables.py (hunk 1 above)
Comment (pierrejeambrun) [reply airflow/api_fastapi/core_api/routes/public/variables.py:174]: Exactly

**Review by pierrejeambrun — COMMENTED** [review]
Thanks

File: airflow/api_fastapi/core_api/routes/public/variables.py (hunk 2 above)
Comment (shubhamraj-git) [reply airflow/api_fastapi/core_api/routes/public/variables.py]: used the same const to keep single source of truth which is 250

File: airflow/api_fastapi/core_api/routes/public/variables.py (hunk 1 above)
Comment (shubhamraj-git) [reply airflow/api_fastapi/core_api/routes/public/variables.py:174]: Done

**Review by jscheffl — APPROVED** [review]

File: airflow/api_fastapi/core_api/routes/public/variables.py [hunk 3]
```diff
@@ -160,12 +160,23 @@ def patch_variable(
 @variables_router.post(
     "",
     status_code=status.HTTP_201_CREATED,
+    responses=create_openapi_http_exception_doc(
+        [status.HTTP_409_CONFLICT, status.HTTP_422_UNPROCESSABLE_ENTITY]
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/variables.py]: 422 is automatically added by fastapi on all endpoints.

We don’t need to add it manually.

**Review by pierrejeambrun — APPROVED** [review]
One small nit, looking good.

File: airflow/api_fastapi/core_api/routes/public/variables.py (hunk 3 above)
Comment (shubhamraj-git) [reply airflow/api_fastapi/core_api/routes/public/variables.py]: Oh, okay

**Comment by jscheffl** [comment]
> One test need fixing.

Ups, had the same comment but now realize at the point coming back that it was pending :face_in_clouds:

**Review by pierrejeambrun — APPROVED** [review]

**Pull request merged**